cst/plugins/eu.esdihumboldt.cst.functions.geometric/src/eu/esdihumboldt/cst/functions/geometric/GeometryHelperFunctions.groovy

```diff
@@ -150,6 +153,69 @@ class GeometryHelperFunctions {
 	}
 
 	/**
+	 * Specification for a _boundaryCovers function
+	 */
+	public static final HelperFunctionSpecification _boundaryCovers_spec = new HelperFunctionSpecification(
+	"Determine if the boundary of a given geometry covers all points of a line.", "true if the all points of the line are on the geometry's boundary",
+	new HelperFunctionArgument("geometry", "Geometry"),
+	new HelperFunctionArgument("line", "Line"),
+	);
+
+	/**
+	 * Determine if the boundary of a geometry covers all points of a line.
+	 *
+	 * @param args the function arguments
+	 * @return true if the geometry's boundary completely covers the line
+	 */
+	@Nullable
+	static boolean _boundaryCovers(Map args) {
+		return args.geometry.getGeometry().getBoundary().covers(args.line.getGeometry());
+	}
+
+	/**
+	 * Specification for the _spatialIndexQuery function
+	 */
+	public static final HelperFunctionSpecification _spatialIndexQuery_spec = new HelperFunctionSpecification(
+	"Query a spatial index. The bounding box of the given geometry is computed and used to query the spatial index.", "Collection of instances that match the query.",
+	new HelperFunctionArgument("spatialIndex",
```

Comment:
Yes, I'd definitely prefer that, too, but I didn't see a way how, since the helper function is static and the `SpatialIndexService` has project scope (so I can't use `HalePlatform.getServiceProvider()`). Is there a way to get the service in a static way so that it works both in the UI and the CLI?